webapp/channels/src/components/widgets/modals/components/modal_sidebar.tsx

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useLayoutEffect} from 'react';
+
+import * as UserAgent from 'utils/user_agent';
+
+import './modal_sidebar.scss';
+
+export type Tab = {
+    icon: JSX.Element;
+    name: string;
+    uiName: string;
+}
+
+export type Props = {
+    activeTab?: string;
+    tabs: Tab[];
+    updateTab: (name: string) => void;
+}
+
+function ModalSidebar({tabs, activeTab, updateTab}: Props) {
+    const handleClick = (tab: Tab, e: React.MouseEvent) => {
+        e.preventDefault();
+        updateTab(tab.name);
+        (e.target as Element).closest('.settings-modal')?.classList.add('display--content');
+    };
+
+    useLayoutEffect(() => {
+        if (UserAgent.isFirefox()) {
+            document.querySelector('.settings-modal .settings-table .nav')?.classList.add('position--top');
```

Comment:
0/5 since this code is coming from the existing settings modal, but I wonder if this is still needed in modern browsers since I'm not sure why we'd set the top for this only on Firefox

Reply:
Actually, is a layout effect the correct place for this? It happens before the browser repaints, so I wouldn't think this would find any elements to add a class to